api-participant/src/main/java/bio/terra/pearl/api/participant/config/B2CConfigurationService.java

```diff
@@ -0,0 +1,99 @@
+package bio.terra.pearl.api.participant.config;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+import org.yaml.snakeyaml.LoaderOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.Constructor;
+
+@Slf4j
+@Service
+public class B2CConfigurationService {
+
+  @Getter private B2CConfiguration b2cConfiguration;
+  private final Map<String, Map<String, String>> portalToConfig = new HashMap<>();
+
+  public B2CConfigurationService() {
+    b2cConfiguration = new B2CConfiguration();
+  }
+
+  /**
+   * Get the B2C configuration for a portal. Returns an empty map if portal config for portal is not
+   * specified
+   */
+  public Map<String, String> getB2CForPortal(String portalShortcode) {
+    // see if already constructed
+    Map<String, String> portalConfig = portalToConfig.get(portalShortcode);
+    if (portalConfig == null) {
+      portalConfig = buildConfigMap(portalShortcode);
+    }
+
+    return portalConfig;
+  }
+
+  /**
+   * Initialize B2C configuration from a yaml file. The file can be specified as an absolute path or
+   * a relative path. If relative, it is loaded from the classpath.
+   */
+  public void initB2CConfig(String b2cConfigFile) {
+    if (StringUtils.isBlank(b2cConfigFile)) {
+      log.error("b2c-config-file property is not set");
+      return;
+    }
+
+    Yaml yaml = new Yaml(new Constructor(B2CConfiguration.class, new LoaderOptions()));
+
+    File file = new File(b2cConfigFile);
```

Comment:
```suggestion
    File file = new File(b2cConfigFile);
    /** an absolute file path will be used in deployed environments as the file will be placed by terraform.  In local dev, we read the config from source */
```

Reply:
With this implementation we can also support a local file at an absolute path, if that becomes our longer term plan.